SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStrategySystem.cpp

```diff
@@ -0,0 +1,125 @@
+#include "Interop/SpatialStrategySystem.h"
+#include "Interop/Connection/SpatialOSWorkerInterface.h"
+#include "Schema/ServerWorker.h"
+#include "Utils/InterestFactory.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialStrategySystem);
+
+namespace SpatialGDK
+{
+void SpatialStrategySystem::Advance(SpatialOSWorkerInterface* Connection)
+{
+	// Messages are inspected to take care of the Worker entity creation.
+	const TArray<Worker_Op>& Messages = Connection->GetWorkerMessages();
+	for (const auto& Message : Messages)
+	{
+		switch (Message.op_type)
+		{
+		case WORKER_OP_TYPE_RESERVE_ENTITY_IDS_RESPONSE:
+		{
+			const Worker_ReserveEntityIdsResponseOp& Op = Message.op.reserve_entity_ids_response;
+			if (Op.request_id == StrategyWorkerRequest)
+			{
+				if (Op.first_entity_id == SpatialConstants::INVALID_ENTITY_ID)
+				{
+					UE_LOG(LogSpatialStrategySystem, Error, TEXT("Reserve entity failed : %s"), UTF8_TO_TCHAR(Op.message));
```

Comment:
Should this be `Fatal`, as the StrategyWorker will be simply stuck in this state? Or maybe we need to add retries?
Another question, possibly addressed at @ImprobableNic, is whether we need to ReserveEntityIds at all? Can we let the Runtime allocate an ID?

Reply:
I do not have a well formed opinion on the matter of reserving or letting the runtime allocate it. I guess it does not make much difference in that specific case.
But we could remove this code if we put the strategy worker entity in the initial snapshot.

Reply:
Spatial view has a retry thing in it. Reserve entity ID can't fail forever.

Reply:
I've changed Strategy Partition Entity to be statically created in the snapshot rather than dynamically.